includes/abstracts/abstract-wc-product.php

```diff
@@ -1420,6 +1420,13 @@ public function is_visible() {
 			$visible = false;
 		}
 
+		$parent_id = $this->get_parent_id();
+		if ( ! empty( $parent_id ) ) {
+			if ( ( $parent_product = wc_get_product( $parent_id ) ) && 'publish' !== $parent_product->get_status() ) {
```

Comment:
This will raise a coding standard issue.

> Assignments must be the first block of code on a line

You should assign the variable before the conditional.

Reply:
Opted to remove the additional variable assignment.